---
title: "R Notebook"
output: html_notebook
---


This script is adapted from Shade and Stopnisek 2019: https://github.com/ShadeLab/PAPER_Shade_CurrOpinMicro

```{r}
library(tidyverse)
library(qiime2R)
library(reshape2)
library(vegan)
library(ggsci)
```


```{r}
nReads=5000                                                          # input dataset needs to be rarefied and the rarefaction depth included 
otu <- read_qza('RedPanda16S/SCBI_redpanda_97otutable_rarefied.qza')$data # read in the rarefied data
map <- read_tsv('RedPanda16S/redpanda_metadata.tsv') # read in the metadata
names(map)[1]<- "sample_id" # rename the sample id column
map<-map[-1,] # remove the QIIME mapping row 
otu_PA <- 1*((otu>0)==1)                                               # presence-absence data

# filter to keep only samples that are from pandas
map<-filter(map, map$Panda_or_Diet == "Panda") %>%
  mutate("sample_id" = gsub("-", ".", .$sample_id)) %>%
  filter(., Geo == "SCBI")

# format OTU table, filter to remove samples that were removed from metadata
otu<-t(otu) %>%
  as.data.frame() %>%
  rownames_to_column("sample_id") %>%
  mutate("sample_id" = gsub("-", ".", .$sample_id)) %>%
  filter(., .$sample_id %in% map$sample_id) %>%
  column_to_rownames(., "sample_id") %>%
  as.matrix() %>%
  t()

otu_PA <- 1*((otu>0)==1)   # presence-absence data
otu_occ <- rowSums(otu_PA)/ncol(otu_PA) # occupancy calculation
otu_rel <- apply(decostand(otu, method="total", MARGIN=2),1, mean)     # mean relative abundance
occ_abun <- data.frame(otu_occ=otu_occ, otu_rel=otu_rel) %>%           # combining occupancy and abundance data frame
  rownames_to_column('otu')

# Occupancy abundance plot:
ggplot(data=occ_abun, aes(x=log10(otu_rel), y=otu_occ)) +
  geom_point(pch=21, fill='white') +
  labs(x="log10(mean relative abundance)", y="Occupancy")

```
```{r}
# Ranking OTUs based on their occupancy
# For calculating ranking index we included following conditions:
#   - time-specific occupancy (sumF) = frequency of detection within time point (genotype or site)
#   - replication consistency (sumG) = has occupancy of 1 in at least one time point (genotype or site) (1 if occupancy 1, else 0)

PresenceSum <- data.frame(otu = as.factor(row.names(otu)), otu) %>% 
  gather(sample_id, abun, -otu) %>%
  left_join(map, by = 'sample_id') %>%
  group_by(otu, Season) %>% # replace Season with time grouping factor 
  summarise(time_freq=sum(abun>0)/length(abun),            # frequency of detection between time points
            coreTime=ifelse(time_freq == 1, 1, 0)) %>%     # 1 only if occupancy 1 with specific time, 0 if not
  group_by(otu) %>%
  summarise(sumF=sum(time_freq),
            sumG=sum(coreTime),
            nS=length(Season),
            Index=(sumF+sumG)/nS)                 # calculating weighting Index based on number of time points detected  


otu_ranked <- occ_abun %>%
  left_join(PresenceSum, by='otu') %>%
  transmute(otu=otu,                           
            rank=Index) %>%
  arrange(desc(rank)) # arranges OTUs based on their occupancy rank (# of time points)

# Calculating the contribution of ranked OTUs to the BC similarity
BCaddition <- NULL

# calculating BC dissimilarity based on the 1st ranked OTU
otu_start=otu_ranked$otu[1]                   
start_matrix <- as.matrix(otu[otu_start,])
start_matrix <- t(start_matrix)
x <- apply(combn(ncol(start_matrix), 2), 2, function(x) sum(abs(start_matrix[,x[1]]- start_matrix[,x[2]]))/(2*nReads))
x_names <- apply(combn(ncol(start_matrix), 2), 2, function(x) paste(colnames(start_matrix)[x], collapse=' - '))
df_s <- data.frame(x_names,x)
names(df_s)[2] <- 1 
BCaddition <- rbind(BCaddition,df_s)
# calculating BC dissimilarity based on addition of ranked OTUs from 2nd to 500th. 
# Can be set to the entire length of OTUs in the dataset, however it might take 
# some time if more than 5000 OTUs are included.
for(i in 2:500){                              
  otu_add=otu_ranked$otu[i]                       
  add_matrix <- as.matrix(otu[otu_add,])
  add_matrix <- t(add_matrix)
  start_matrix <- rbind(start_matrix, add_matrix)
  x <- apply(combn(ncol(start_matrix), 2), 2, function(x) sum(abs(start_matrix[,x[1]]-start_matrix[,x[2]]))/(2*nReads))
  x_names <- apply(combn(ncol(start_matrix), 2), 2, function(x) paste(colnames(start_matrix)[x], collapse=' - '))
  df_a <- data.frame(x_names,x)
  names(df_a)[2] <- i 
  BCaddition <- left_join(BCaddition, df_a, by=c('x_names'))
}
# calculating the BC dissimilarity of the whole dataset (not needed if the second loop is already including all OTUs) 
x <-  apply(combn(ncol(otu), 2), 2, function(x) sum(abs(otu[,x[1]]-otu[,x[2]]))/(2*nReads))   
x_names <- apply(combn(ncol(otu), 2), 2, function(x) paste(colnames(otu)[x], collapse=' - '))
df_full <- data.frame(x_names,x)
names(df_full)[2] <- length(rownames(otu))
BCfull <- left_join(BCaddition,df_full, by='x_names')

rownames(BCfull) <- BCfull$x_names
temp_BC <- BCfull
temp_BC$x_names <- NULL
temp_BC_matrix <- as.matrix(temp_BC)

BC_ranked <- data.frame(rank = as.factor(row.names(t(temp_BC_matrix))),t(temp_BC_matrix)) %>% 
  gather(comparison, BC, -rank) %>%
  group_by(rank) %>%
  summarise(MeanBC=mean(BC)) %>%            # mean Bray-Curtis dissimilarity
  arrange(desc(-MeanBC)) %>%
  mutate(proportionBC=MeanBC/max(MeanBC))   # proportion of the dissimilarity explained by the n number of ranked OTUs
Increase=BC_ranked$MeanBC[-1]/BC_ranked$MeanBC[-length(BC_ranked$MeanBC)]
increaseDF <- data.frame(IncreaseBC=c(0,(Increase)), rank=factor(c(1:(length(Increase)+1))))
BC_ranked <- left_join(BC_ranked, increaseDF)
BC_ranked <- BC_ranked[-nrow(BC_ranked),]

#Creating thresholds for core inclusion - elbow and last call method

#Method: 
#A) Elbow method (first order difference) (script modified from https://pommevilla.github.io/random/elbows.html)
fo_difference <- function(pos){
  left <- (BC_ranked[pos, 2] - BC_ranked[1, 2]) / pos
  right <- (BC_ranked[nrow(BC_ranked), 2] - BC_ranked[pos, 2]) / (nrow(BC_ranked) - pos)
  return(left - right)
}
BC_ranked$fo_diffs <- sapply(1:nrow(BC_ranked), fo_difference)

elbow <- which.max(BC_ranked$fo_diffs)

#B) Final increase in BC similarity of equal or greater then 2% 
lastCall <- last(as.numeric(as.character(BC_ranked$rank[(BC_ranked$IncreaseBC>=1.02)])))
#Creating plot of Bray-Curtis similarity
ggplot(BC_ranked[1:100,], aes(x=factor(BC_ranked$rank[1:100], levels=BC_ranked$rank[1:100]))) +
  geom_point(aes(y=proportionBC)) +
  theme_classic() + theme(strip.background = element_blank(),axis.text.x = element_text(size=7, angle=45)) +
  geom_vline(xintercept=elbow, lty=3, col='red', cex=.5) +
  geom_vline(xintercept=lastCall, lty=3, col='blue', cex=.5) +
  labs(x='ranked OTUs',y='Bray-Curtis similarity') +
  annotate(geom="text", x=elbow+14, y=.1, label=paste("Elbow method"," (",elbow,")", sep=''), color="red")+    
  annotate(geom="text", x=lastCall+3, y=.5, label=paste("Last 2% increase (",lastCall,")",sep=''), color="blue")

occ_abun$fill <- 'no'
occ_abun$fill[occ_abun$otu %in% otu_ranked$otu[1:lastCall]] <- 'core_BC'
occ_abun$elbow <- 'no'
occ_abun$elbow[occ_abun$otu %in% otu_ranked$otu[1:as.numeric(elbow)]] <- 'core_elbow'
```

```{r Determine Core Taxa by Joining with Taxonomy}

taxonomy<-read_qza("RedPanda16S/redpanda_taxonomy.qza")$data
joined_elbow<-filter(taxonomy, taxonomy$Feature.ID %in% filter(occ_abun, occ_abun$elbow == "core_elbow")$otu) %>%
  left_join(., occ_abun, c("Feature.ID"="otu" ))
#write.csv(joined_RRC_longitudinal_elbow, "core_taxa_longitudinal_RRC_elbowmethod_redo.csv")
joined_bc<-filter(taxonomy, taxonomy$Feature.ID %in% filter(occ_abun, occ_abun$fill == "core_BC")$otu)%>%
  left_join(., occ_abun, c("Feature.ID"="otu" ))
#write.csv(joined_RRC_longitudinal_bc, "core_taxa_longitudinal_RRC_BC_redo.csv")
```


```{r determining if presnt in diet}
otu <- read_qza('16S/psychro-mito-chloro-cyano-removed-pet-table.qza')$data # read in the rarefied data
map <- read_tsv('RedPandaMetadataValidated.tsv') # read in the metadata
names(map)[1]<- "sample_id" # rename the sample id column
map<-map[-1,] # remove the QIIME mapping row 

# filter to keep only samples that are from pandas
map.diet<-filter(map, map$Panda_or_Diet == "Diet") %>%
  mutate("sample_id" = gsub("-", ".", .$sample_id))

# format OTU table, filter to remove samples that were removed from metadata
otu.diet<-t(otu) %>%
  as.data.frame() %>%
  rownames_to_column("sample_id") %>%
  mutate("sample_id" = gsub("-", ".", .$sample_id)) %>%
  filter(., .$sample_id %in% map.diet$sample_id)


bc.present.in.diet <- otu.diet %>%
  gather(., "Feature.ID", "abundance", -sample_id) %>%
  group_by(Feature.ID) %>%
  summarize(counts = sum(abundance)) %>%
  filter(.$Feature.ID %in% joined_bc$Feature.ID) %>%
  left_join(joined_bc, ., by = "Feature.ID")

bc.present.in.diet$present.in.diet<- ifelse(bc.present.in.diet$counts > 0, "True", "False")
occ_abun.core <- occ_abun %>%
  mutate("Feature.ID" = otu) %>%
  left_join(bc.present.in.diet, ., by = "Feature.ID")
write_csv(occ_abun.core, "core_microbiome_final_occupancysabundance.csv")

elbow.present.in.diet <- otu.diet %>%
  gather(., "Feature.ID", "abundance", -sample_id) %>%
  group_by(Feature.ID) %>%
  summarize(counts = sum(abundance)) %>%
  filter(.$Feature.ID %in% joined_elbow$Feature.ID) %>%
  left_join(., joined_elbow, by = "Feature.ID")
```

